README.md
# freetds-ios


freetds-1.00.61/ChangeLog
Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Sun Aug 7 10:57:57 2016 +0100

    Fix small memory leak

    Silly mistake caused a duplicate line.

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Fri Jul 22 11:23:50 2016 +0100

    Reduce indentation level in pool_user_read

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Fri Jul 22 11:25:24 2016 +0100

    Remove warning compiling with clang

    clang complains about signed shift not portable.

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Fri Jul 22 10:27:08 2016 +0100

    Remove some string overflows from VMS code

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Fri Jul 22 10:29:19 2016 +0100

    pool: Avoid to overwrite packet if not entirely written

    This can happen if packet is not entirely written to server after
    being read from client.

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Mon Jul 18 12:09:07 2016 +0100

    Extends cs_dt_crack to support newer types

    This add a new cs_dt_crack_v2 ABI function (cs_dt_crack API is redirected
    to it using a preprocessor macro). This to make the extension of
    CS_DATEREC more safe (version of 1.0 could call cs_dt_crack with
    CS_BIGDATETIME_TYPE/CS_BIGTIME_TYPE without having these fields written
    which would possible cause memory corruption on version compiled with
    these fields missing).
    This make sqsh compile and work with newer date/time types.

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Mon Jul 18 09:38:13 2016 +0100

    Make AppVeyor build with OpenSSL enabled

    CMake was not finding OpenSSL.

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Tue Jun 21 21:45:21 2016 +0100

    attempt multiple connections if availability groups are detected

    This allows the server to start up properly.
    Connection are retried after 1 seconds for maximum 10 times.
    The global timeout still is in place.

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Mon Jun 27 12:59:07 2016 +0100

    Fix NUMERIC column for bcp native format

    Prefix was not handled correctly.

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Sun Jun 26 08:30:08 2016 +0100

    Fix freebcp -E (identity columns)

    An additional query prevented code to handle correctly column types.

Author: Manfroi Fabrice <fm@4js.com>
Date:   Wed Jun 22 16:33:53 2016 +0200

    Fix build error on AIX 6.1 / 7.1 / 7.2

    tls.c:719: error: 'AF_INET6' undeclared (first use in this function)
    tls.c:719: error: (Each undeclared identifier is reported only once
    tls.c:719: error: for each function it appears in.)
    tls.c:722: error: 'AF_INET' undeclared (first use in this function)

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Tue Jun 21 21:41:23 2016 +0100

    enable again tds_gettime_ms

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Tue Jun 21 21:38:03 2016 +0100

    fix possible file descriptor leak

    This was introduced quite recently

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Mon Jun 20 14:46:57 2016 +0100

    attempt to connect to any address at a time to support availability groups

    If availability groups are available the servers may be listening for
    connection based on state.
    Try all IPs to get the good one. The attempts are made in parallel.
    There is still no retry to wait a server available on switch time.

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Sat Jun 18 21:08:17 2016 +0100

    Fix internal result state in dbcanquery

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Sat Jun 18 21:09:17 2016 +0100

    Test dbcanquery behaviour

    Function should discard all row results.

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Sat Jun 18 13:27:56 2016 +0100

    Avoid to strip spaces from procedure text.

    NTBSTRINGBIND strip spaces at the end.
    Using STRINGBIND with 0 size cause string to be just NUL terminated.
    The 0 would lead to a possible overflow however we check the column
    len before doing the bind to avoid such possibility.

    Original patch from Nem W Schlecht.

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Fri Jun 17 22:33:24 2016 +0100

    Remove redundant check in dbcanquery

    Check already in CHECK_CONN.

Author: Nem W Schlecht <nem@emptec.com>
Date:   Fri Jun 17 22:38:12 2016 +0100

    Avoid crash on connection error and report an error instead

    Patch from Nem W Schlecht.

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Sun Jun 12 21:14:30 2016 +0100

    remove old ChangeLog files

    ChangeLogs were not updated, now git log is used

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Sat Jun 11 09:33:56 2016 +0100

    remove ODBCNODMLIBAPP automake macro

    Avoid using two macros for the same purpose using absolute path.

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Sun Jun 12 21:09:43 2016 +0100

    allows to specify 5.0 as tds protocol version during configure

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Sat Jun 11 09:32:43 2016 +0100

    do not compile as bundle under Mac OS X is --with-odbc-nodm is specified

    In this case the driver will be used as a dynamic library (not loaded
    dynamically) so compile as such.
    In case a DM is used unit tests/application will link to the DM which
    will load the driver (requiring to be a module on old Mac OS X versions).

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Fri Jun 10 11:45:44 2016 +0100

    replace BYTE with unsigned char in ODBC files

    BYTE is not supported on some platforms.
    This fixes Mac OS X build.
    unixODBC headers files and Windows platform define it.

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Sun Jun 5 00:20:06 2016 +0100

    Add some missing include to fisql utility

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Sun Jun 5 00:06:55 2016 +0100

    Support all types in dbvarylen

Author: Frediano Ziglio <freddy77@gmail.com>
Date:   Sun Jun 5 00:04:46 2016 +0100


freetds-1.00.61/src/tds/tds_willconvert.h
/*** Please do not edit this file!                                       ***/
/*** It was generated with 'perl tds_willconvert.pl > tds_willconvert.h' ***/
/*** It is much easier to edit the __DATA__ table than this file.        ***/
/***                                                                     ***/
/***                                                          Thank you. ***/

static const uint8_t type2category[256] = {
	0, /* 0 */
	0, /* 1 */
	0, /* 2 */
	0, /* 3 */
	0, /* 4 */
	0, /* 5 */
	0, /* 6 */
	0, /* 7 */
	0, /* 8 */
	0, /* 9 */
	0, /* 10 */
	0, /* 11 */
	0, /* 12 */
	0, /* 13 */
	0, /* 14 */
	0, /* 15 */
	0, /* 16 */
	0, /* 17 */
	0, /* 18 */
	0, /* 19 */
	0, /* 20 */
	0, /* 21 */
	0, /* 22 */
	0, /* 23 */
	0, /* 24 */
	0, /* 25 */
	0, /* 26 */
	0, /* 27 */
	0, /* 28 */
	0, /* 29 */
	0, /* 30 */
	0, /* SYBVOID */
	0, /* 32 */
	0, /* 33 */
	1, /* SYBIMAGE */
	4, /* SYBTEXT */
	5, /* SYBBLOB */
	1, /* SYBVARBINARY */
	0, /* SYBINTN */
	4, /* SYBVARCHAR */
	3, /* SYBMSDATE */
	3, /* SYBMSTIME */
	3, /* SYBMSDATETIME2 */
	3, /* SYBMSDATETIMEOFFSET */
	0, /* 44 */
	1, /* SYBBINARY */
	0, /* SYBINTERVAL */
	4, /* SYBCHAR */
	2, /* SYBINT1 */
	3, /* SYBDATE */
	7, /* SYBBIT */
	3, /* SYBTIME */
	2, /* SYBINT2 */
	0, /* 53 */
	0, /* 54 */
	0, /* 55 */
	2, /* SYBINT4 */
	0, /* 57 */
	3, /* SYBDATETIME4 */
	2, /* SYBREAL */
	2, /* SYBMONEY */
	3, /* SYBDATETIME */
	2, /* SYBFLT8 */
	0, /* 63 */
	2, /* SYBUINT1 */
	2, /* SYBUINT2 */
	2, /* SYBUINT4 */
	2, /* SYBUINT8 */
	0, /* SYBUINTN */
	0, /* 69 */
	0, /* 70 */
	0, /* 71 */
	0, /* 72 */
	0, /* 73 */
	0, /* 74 */
	0, /* 75 */
	0, /* 76 */
	0, /* 77 */
	0, /* 78 */
	0, /* 79 */
	0, /* 80 */
	0, /* 81 */
	0, /* 82 */
	0, /* 83 */
	0, /* 84 */
	0, /* 85 */
	0, /* 86 */
	0, /* 87 */
	0, /* 88 */
	0, /* 89 */
	0, /* 90 */
	0, /* 91 */
	0, /* 92 */
	0, /* 93 */
	0, /* 94 */
	0, /* 95 */
	0, /* 96 */
	0, /* 97 */
	0, /* SYBVARIANT */
	0, /* SYBNTEXT */
	0, /* 100 */
	0, /* 101 */
	0, /* 102 */
	6, /* SYBSENSITIVITY */
	7, /* SYBBOUNDARY */
	0, /* 105 */
	7, /* SYBDECIMAL */
	0, /* 107 */
	7, /* SYBNUMERIC */
	0, /* SYBFLTN */
	0, /* SYBMONEYN */
	0, /* SYBDATETIMN */
	0, /* 112 */
	0, /* 113 */
	0, /* 114 */
	0, /* 115 */
	0, /* 116 */
	0, /* 117 */
	0, /* 118 */
	0, /* 119 */
	0, /* 120 */
	0, /* 121 */
	2, /* SYBMONEY4 */
	0, /* SYBDATEN */
	0, /* 124 */
	0, /* 125 */
	0, /* 126 */
	2, /* SYBINT8 */
	0, /* 128 */
	0, /* 129 */
	0, /* 130 */
	0, /* 131 */
	0, /* 132 */
	0, /* 133 */
	0, /* 134 */
	0, /* 135 */
	0, /* 136 */
	0, /* 137 */
	0, /* 138 */
	0, /* 139 */
	0, /* 140 */
	0, /* 141 */
	0, /* 142 */
	0, /* 143 */
	0, /* 144 */
	0, /* 145 */
	0, /* 146 */
	0, /* SYBTIMEN */
	0, /* 148 */
	0, /* 149 */
	0, /* 150 */
	0, /* 151 */
	0, /* 152 */
	0, /* 153 */
	0, /* 154 */
	0, /* 155 */
	0, /* 156 */
	0, /* 157 */
	0, /* 158 */
	0, /* 159 */
	0, /* 160 */
	0, /* 161 */
	0, /* 162 */
	0, /* SYBXML */
	0, /* 164 */
	1, /* XSYBVARBINARY */
	0, /* 166 */
	4, /* XSYBVARCHAR */
	0, /* 168 */
	0, /* 169 */
	0, /* 170 */
	0, /* 171 */
	0, /* 172 */
	1, /* XSYBBINARY */
	0, /* SYBUNITEXT */
	4, /* SYBLONGCHAR */
	0, /* SYBSINT1 */
	0, /* 177 */
	0, /* 178 */
	0, /* 179 */
	0, /* 180 */
	0, /* 181 */
	0, /* 182 */
	0, /* 183 */
	0, /* 184 */
	0, /* 185 */
	0, /* 186 */
	3, /* SYB5BIGDATETIME */
	3, /* SYB5BIGTIME */
	0, /* 189 */
	0, /* 190 */
	0, /* SYB5INT8 */
	0, /* 192 */
	0, /* 193 */
	0, /* 194 */
	0, /* 195 */
	0, /* 196 */
	0, /* 197 */
	0, /* 198 */
	0, /* 199 */
	0, /* 200 */
	0, /* 201 */
	0, /* 202 */
	0, /* 203 */
	0, /* 204 */
	0, /* 205 */
	0, /* 206 */
	0, /* 207 */
	0, /* 208 */
	0, /* 209 */
	0, /* 210 */
	0, /* 211 */
	0, /* 212 */
	0, /* 213 */
	0, /* 214 */
	0, /* 215 */
	0, /* 216 */
	0, /* 217 */
	0, /* 218 */
	0, /* 219 */
	0, /* 220 */
	0, /* 221 */
	0, /* 222 */
	0, /* 223 */
	0, /* 224 */
	1, /* SYBLONGBINARY */
	0, /* 226 */
	0, /* 227 */
	0, /* 228 */
	0, /* 229 */
	0, /* 230 */
	0, /* XSYBNVARCHAR */
	0, /* 232 */
	0, /* 233 */
	0, /* 234 */
	0, /* 235 */
	0, /* 236 */
	0, /* 237 */
	0, /* 238 */
	0, /* XSYBNCHAR */
	0, /* SYBMSUDT */
	0, /* SYBMSXML */
	0, /* 242 */
	0, /* 243 */
	0, /* 244 */
	0, /* 245 */
	0, /* 246 */
	0, /* 247 */
	0, /* 248 */
	0, /* 249 */
	0, /* 250 */
	0, /* 251 */
	0, /* 252 */
	0, /* 253 */
	0, /* 254 */
	0, /* 255 */
};

static const uint32_t category_conversion[] = {
	0,
	22,	/* SYBIMAGE */
	150,	/* SYBINT1 */
	26,	/* SYBMSDATE */
	190,	/* SYBTEXT */
	50,	/* SYBUNIQUE */
	0,	/* SYBSENSITIVITY */
	150,	/* SYBBIT */
};
